### PR TITLE
Changes to Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk needed for repo consolidation

### PR DESCRIPTION
**Pull request by jkoritzinsky**
In a consolidated repo, the files to support creating the Microsoft.NETCore.App shared framework are not going to be in the root of the repository. Additionally, `ProjectDir` is not always equal to `RepoRoot`.

These changes enable a custom license file location and a custom signing projects/tools directory.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <Project>
 
+  <PropertyGroup>
+    <SigningToolsDirectory Condition="'$(SigningToolsDirectory)' != ''">$(RepoRoot)signing\</SigningToolsDirectory>
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props]: This is the convention I follow:

```suggestion
    <SigningToolsDir Condition="'$(SigningToolsDir)' != ''">$(RepoRoot)signing\</SigningToolsDir>
```

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+    <LicenseFileLocation Condition="'$(LicenseFileLocation)' != ''">$(RepoRoot)LICENSE.TXT</LicenseFileLocation>
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props]: Maybe not exactly this name if it collides, but `File` being the suffix for a file path is the style here:

```suggestion
    <LicenseFile Condition="'$(LicenseFile)' != ''">$(RepoRoot)LICENSE.TXT</LicenseFile>
```

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/windows/wix.targets [hunk 3]
```diff
@@ -194,7 +194,7 @@
     the bundle references.
   -->
   <Target Name="GetBundledMsiFiles">
-    <MSBuild Projects="$(RepoRoot)signing\SignMsiFiles.proj" Targets="Build" />
+    <MSBuild Projects="$(SigningToolsDirectory)\SignMsiFiles.proj" Targets="Build" />
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/windows/wix.targets]: Redundant `\` (and applying `Dir` change):

```suggestion
    <MSBuild Projects="$(SigningToolsDir)SignMsiFiles.proj" Targets="Build" />
```

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets [hunk 4]
```diff
@@ -251,7 +251,7 @@
     generate shared frameworks in the Arcade SDK.
   -->
   <Target Name="BuildPkgProjectReferences">
-    <MSBuild Projects="$(RepoRoot)signing\SignBinaries.proj" Targets="Build" />
+    <MSBuild Projects="$(SigningToolsDirectory)\SignBinaries.proj" Targets="Build" />
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets]: Redundant `\` (and applying `Dir` change):

```suggestion
    <MSBuild Projects="$(SigningToolsDir)\SignBinaries.proj" Targets="Build" />
```

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/installer.targets [hunk 5]
```diff
@@ -279,7 +279,7 @@
   -->
   <Target Name="GetCommonJsonProperties">
     <PropertyGroup>
-      <FullLicenseText>$([System.IO.File]::ReadAllText('$(ProjectDir)LICENSE.TXT').Replace('%0A', '\n').Replace('"', '\"'))</FullLicenseText>
+      <FullLicenseText>$([System.IO.File]::ReadAllText('$(LicenseFileLocation)').Replace('%0A', '\n').Replace('"', '\"'))</FullLicenseText>
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/installer.targets]: Applying `File` change:

```suggestion
      <FullLicenseText>$([System.IO.File]::ReadAllText('$(LicenseFile)').Replace('%0A', '\n').Replace('"', '\"'))</FullLicenseText>
```

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props [hunk 6]
```diff
@@ -1,5 +1,10 @@
 <Project>
 
+  <PropertyGroup>
+    <SigningToolsDir Condition="'$(SigningToolsDir)' != ''">$(RepoRoot)signing\</SigningToolsDir>
+    <LicenseFile Condition="'$(LicenseFile)' != ''">$(RepoRoot)LICENSE.TXT</LicenseFile>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props]: Those two conditions should be inverted to only set the default if the property is unset.

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets [hunk 7]
```diff
@@ -251,7 +251,7 @@
     generate shared frameworks in the Arcade SDK.
   -->
   <Target Name="BuildPkgProjectReferences">
-    <MSBuild Projects="$(RepoRoot)signing\SignBinaries.proj" Targets="Build" />
+    <MSBuild Projects="$(SigningToolsDir)SignBinaries.proj" Targets="Build" />
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets]: When you don't control all the properties yourself the MSBuild helper functions are useful:
```suggestion
    <MSBuild Projects="$([MSBuild]::NormalizePath('$(SigningToolsDir)', 'SignBinaries.proj'))" Targets="Build" />
```

Comment (ViktorHofer) [reply src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets]: This makes sure that directory seperators are placed correctly if missing.

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/windows/wix.targets [hunk 8]
```diff
@@ -194,7 +194,7 @@
     the bundle references.
   -->
   <Target Name="GetBundledMsiFiles">
-    <MSBuild Projects="$(RepoRoot)signing\SignMsiFiles.proj" Targets="Build" />
+    <MSBuild Projects="$(SigningToolsDir)SignMsiFiles.proj" Targets="Build" />
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/windows/wix.targets]: Same here:
```suggestion
    <MSBuild Projects="$([MSBuild]::NormalizeDirectory('$(SigningToolsDir)', 'SignMsiFiles.proj'))" Targets="Build" />
```

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props (hunk 6 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props]: and should probably could use the msbuild helper functions as well. For the first one you would use NormalizeDirectory (without the slash at the end) and for the second one the NormalizePath one again.

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets (hunk 7 above)
Comment (dagood) [reply src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.sharedfx.targets]: Interesting, haven't seen this before. Unfortunate how wordy it is, but nice.

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props (hunk 6 above)
Comment (jkoritzinsky) [reply src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.props]: Good catch! I'll go fix it.

**Pull request merged**